--- dpdk-stable-21.11.2/usertools/dpdk-devbind.py	2022-08-29 19:27:14.000000000 +0530
+++ mcdma-dpdk-stable-21.11.2/usertools/dpdk-devbind.py	2022-11-02 19:24:21.551033614 +0530
@@ -62,6 +62,8 @@ intel_idxd_spr = {'Class': '08', 'Vendor
                   'SVendor': None, 'SDevice': None}
 intel_ntb_skx = {'Class': '06', 'Vendor': '8086', 'Device': '201c',
                  'SVendor': None, 'SDevice': None}
+intel_mcdma = {'Class': '06', 'Vendor': '8086', 'Device': '1172',
+              'SVendor': None, 'SDevice': None}
 intel_ntb_icx = {'Class': '06', 'Vendor': '8086', 'Device': '347e',
                  'SVendor': None, 'SDevice': None}
 
@@ -72,7 +74,7 @@ octeontx2_npa = {'Class': '08', 'Vendor'
 octeontx2_ree = {'Class': '08', 'Vendor': '177d', 'Device': 'a0f4',
                  'SVendor': None, 'SDevice': None}
 
-network_devices = [network_class, cavium_pkx, avp_vnic, ifpga_class]
+network_devices = [network_class, cavium_pkx, avp_vnic, ifpga_class, intel_mcdma]
 baseband_devices = [acceleration_class]
 crypto_devices = [encryption_class, intel_processor_class]
 dma_devices = [cnxk_dma, hisilicon_dma,
@@ -279,6 +281,8 @@ def device_type_match(dev, devices_type)
         param_count = len(
             [x for x in devices_type[i].values() if x is not None])
         match_count = 0
+        if dev["Vendor"] == "1172" :
+           return True
         if dev["Class"][0:2] == devices_type[i]["Class"]:
             match_count = match_count + 1
             for key in devices_type[i].keys():
