--- dpdk-stable-21.11.2/drivers/net/meson.build	2022-08-29 19:27:14.000000000 +0530
+++ mcdma-dpdk-stable-21.11.2/drivers/net/meson.build	2022-10-25 18:51:04.945738983 +0530
@@ -34,6 +34,7 @@ drivers = [
         'ixgbe',
         'kni',
         'liquidio',
+        'mcdma',
         'memif',
         'mlx4',
         'mlx5',
--- dpdk-stable-21.11.2/examples/meson.build	2022-08-29 19:27:14.000000000 +0530
+++ mcdma-dpdk-stable-21.11.2/examples/meson.build	2022-10-25 20:28:02.525566721 +0530
@@ -36,6 +36,8 @@ all_examples = [
         'l3fwd-graph',
         'l3fwd-power',
         'link_status_interrupt',
+#      	'mcdma-test/sample',
+	'mcdma-test/perfq',
         'multi_process/client_server_mp/mp_client',
         'multi_process/client_server_mp/mp_server',
         'multi_process/hotplug_mp',
@@ -128,7 +130,7 @@ foreach example: examples
     if allow_experimental_apis
         cflags += '-DALLOW_EXPERIMENTAL_API'
     endif
-    executable('dpdk-' + name, sources,
+    executable('mcdma-test/perfq/build/mcdma-test', sources,
             include_directories: includes,
             link_whole: link_whole_libs,
             link_args: ldflags,
--- dpdk-stable-21.11.2/kernel/linux/meson.build	2022-08-29 19:27:14.000000000 +0530
+++ mcdma-dpdk-stable-21.11.2/kernel/linux/meson.build	2022-11-09 16:27:25.057708312 +0530
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright(c) 2018 Intel Corporation
 
-subdirs = ['kni']
+subdirs = ['kni', 'igb_uio']
 
 kernel_build_dir = get_option('kernel_dir')
 kernel_source_dir = get_option('kernel_dir')
