diff -pNuar dpdk-stable-pmd/app/test-pmd/meson.build dpdk-stable/app/test-pmd/meson.build
--- dpdk-stable-pmd/app/test-pmd/meson.build	2022-11-10 11:17:34.754115349 +0530
+++ dpdk-stable/app/test-pmd/meson.build	2022-11-10 11:02:24.366142306 +0530
@@ -4,6 +4,8 @@
 # override default name to drop the hyphen
 name = 'testpmd'
 cflags += '-Wno-deprecated-declarations'
+includes += include_directories('../../../drivers/net/mcdma/')
+includes += include_directories('../../../drivers/net/mcdma/base/')
 sources = files(
         '5tswap.c',
         'cmdline.c',
@@ -26,6 +28,7 @@ sources = files(
         'testpmd.c',
         'txonly.c',
         'util.c',
+        'ifc_mcdma_testpmd.c'
 )
 
 if dpdk_conf.has('RTE_HAS_JANSSON')
diff -pNuar dpdk-stable-pmd/app/test-pmd/testpmd.c dpdk-stable/app/test-pmd/testpmd.c
--- dpdk-stable-pmd/app/test-pmd/testpmd.c	2022-11-10 11:18:00.202114596 +0530
+++ dpdk-stable/app/test-pmd/testpmd.c	2022-11-10 11:11:51.490125513 +0530
@@ -722,6 +722,9 @@ uint16_t gso_max_segment_size = RTE_ETHE
 /* Holds the registered mbuf dynamic flags names. */
 char dynf_names[64][RTE_MBUF_DYN_NAMESIZE];
 
+#ifdef IFC_QDMA_DYN_CHAN
+extern uint16_t mcdma_ph_chno[AVST_MAX_NUM_CHAN];
+#endif
 
 /*
  * Helper function to check if socket is already discovered.
@@ -2929,7 +2932,7 @@ start_port(portid_t pid)
 						port->socket_id,
 						&(port->txq[qi].conf));
 
-				if (diag == 0) {
+				if (diag >= 0) {
 					port->txq[qi].state =
 						conf->tx_deferred_start ?
 						RTE_ETH_QUEUE_STATE_STOPPED :
@@ -2986,7 +2989,11 @@ start_port(portid_t pid)
 					     &(port->rxq[qi].conf),
 					     mp);
 				}
-				if (diag == 0)
+#ifdef IFC_QDMA_DYN_CHAN
+				/* Get the physical channel number */
+				mcdma_ph_chno[qi] = diag;
+#endif
+				if (diag >= 0)
 					continue;
 
 				/* Fail to setup rx queue, return */
@@ -3050,6 +3057,15 @@ start_port(portid_t pid)
 		need_check_link_status = 1;
 
 		pl[cfg_pi++] = pi;
+#ifndef IFC_BUF_REUSE
+#ifndef IFC_AVST_MULTI_PORT
+#ifdef IFC_AVST_PACKETGEN
+		ifc_single_port_pio_init(pi);
+#endif//IFC_AVST_PACKETGEN
+#else
+		ifc_multi_port_pio_init(pi);
+#endif//IFC_AVST_MULTI_PORT
+#endif//IFC_BUF_REUSE
 	}
 
 	if (need_check_link_status == 1 && !no_link_check)
@@ -3144,6 +3160,13 @@ stop_port(portid_t pid)
 			port->port_status = RTE_PORT_HANDLING;
 		else
 			continue;
+#ifndef IFC_BUF_REUSE
+#ifndef IFC_AVST_MULTI_PORT
+#ifdef IFC_AVST_PACKETGEN
+		ifc_mcdma_pio_cleanup(pi);
+#endif
+#endif
+#endif
 
 		if (hairpin_mode & 0xf) {
 			int j;
diff -pNuar dpdk-stable-pmd/app/test-pmd/testpmd.h dpdk-stable/app/test-pmd/testpmd.h
--- dpdk-stable-pmd/app/test-pmd/testpmd.h	2022-11-10 11:18:00.202114596 +0530
+++ dpdk-stable/app/test-pmd/testpmd.h	2022-11-10 10:58:20.398149530 +0530
@@ -17,6 +17,7 @@
 #endif
 #include <rte_os_shim.h>
 #include <cmdline.h>
+#include "ifc_mcdma_testpmd.h"
 #include <sys/queue.h>
 #ifdef RTE_HAS_JANSSON
 #include <jansson.h>
