--- dpdk-stable/lib/eal/include/rte_interrupts_orig.h	2024-04-01 15:05:12.310501468 -0400
+++ dpdk-stable/lib/eal/include/rte_interrupts.h	2024-04-01 15:02:05.731498549 -0400
@@ -35,7 +35,7 @@ struct rte_intr_handle;
 /** Interrupt instance will be shared between primary and secondary processes. */
 #define RTE_INTR_INSTANCE_F_SHARED      RTE_BIT32(0)
 
-#define RTE_MAX_RXTX_INTR_VEC_ID      512
+#define RTE_MAX_RXTX_INTR_VEC_ID      2048
 #define RTE_INTR_VEC_ZERO_OFFSET      0
 #define RTE_INTR_VEC_RXTX_OFFSET      1
 
